### ISSUE TITLE
Type inheritance enum compilation issue

### ISSUE THREAD

**Issue by thareh** [bmx-ng/bcc#623]
Good day,

The following module will not compile:
```
SuperStrict

Module Test.Test

Framework BRL.Blitz
Import Text.Encoding

Type TFoo
	
	Method New(test:String, e:EStreamEncoding=Null)
	EndMethod
	
EndType

Type TBar Extends TFoo
	
EndType
```

"Unable to convert from Int to Enum EStreamEncoding."

Thanks!

**Comment by woollybah** [comment]
It's not a very good error message. It probably wants to be something else, like "Invalid default enum value"...

**Comment by thareh** [comment]
So it should not compile?

It seems to work fine when not used within a module.

**Comment by GWRon** [comment]
Can enums default to...null?

Am 30. Mai 2023 14:50:53 MESZ schrieb Carl Husberg ***@***.***>:

**Comment by thareh** [comment]
Yes, but it just seems to use the first value in that case. (I thought perhaps it worked like an object and could really be null)
